### PR TITLE
Linux Founder instant answer

### PR DESCRIPTION
**Pull request by dhuette13**
# Goodie Pull Request Template

Using this template will help us better understand your Instant Answer and assist you when necessary.  Simply copy and paste the markdown below into the description of your GitHub pull request and complete as appropriate.

**What does your Instant Answer do?**
It answers the question of who the founder of linux is.

**What problem does your Instant Answer solve (Why is it better than organic links)?**
It gives a straight forward answer at the top of the screen, easily accessible for people interested.

**What is the data source for your Instant Answer? (Provide a link if possible)**
Wikipedia
https://en.wikipedia.org/wiki/Linus_Torvalds

**Why did you choose this data source?**
Accurate information.

**Are there any other alternative (better) data sources?**
N / A

**What are some example queries that trigger this Instant Answer?**
linux founder
founder of linux
founder of linux operating system

**Which communities will this Instant Answer be especially useful for? (gamers, book lovers, etc)**
geeks / computer enthusiasts

**Is this Instant Answer connected to a DuckDuckHack [Instant Answer idea](https://duck.co/ideas)?**
Yes, I saw the request on the idea forum. 
https://duck.co/ideas/idea/4824/answer-to-linux-founder

**Which existing Instant Answers will this one supercede/overlap with?**
Maybe the wikipedia about answer.

**Are you having any problems? Do you need our help with anything?**
This is my first instant answer so I picked something easy. I'm not sure if I did everything correctly
so feedback on my work would be greatly appreciated.

**Where did you hear about DuckDuckHack? (For first time contributors)**
I found it on twitter.

**What does the Instant Answer look like? (Provide a screenshot for new or updated Instant Answers)**

selection_071

## Checklist
Please place an 'X' where appropriate.

```
[X] Added metadata and attribution information (https://duck.co/duckduckhack/metadata)
[X] Wrote test file and added to t/ directory (https://duck.co/duckduckhack/goodie_tests)
[X] Verified that instant answer adheres to design guidelines (https://duck.co/duckduckhack/design_styleguide)
[X] Verified that instant answer adheres to code styleguide (https://duck.co/duckduckhack/code_styleguide)
[X] Tested cross-browser compatibility

    Please let us know which browsers/devices you've tested on:
    - Windows 8
        [] Google Chrome
        [] Firefox
        [] Opera
        [] IE 10

    - Windows 7
        [] Google Chrome
        [X] Firefox
        [] Opera
        [] IE 8
        [] IE 9
        [] IE 10

    - Windows XP
        [] IE 7
        [] IE 8

    - Mac OSX
        [] Google Chrome
        [] Firefox
        [] Opera
        [] Safari

    - iOS (iPhone)
        [] Safari Mobile
        [] Google Chrome
        [] Opera

    - iOS (iPad)
        [] Safari Mobile
        [] Google Chrome
        [] Opera

    - Android
        [] Firefox
        [] Native Browser
        [] Google Chrome
        [] Opera
    - Linux Mint
        [X] Firefox
        [X] Chromium
        [X] Opera
```

### CODE REVIEW

**Comment by javathunderman** [comment]
@dhuette13 Yeah, I heard about this. Happy to see this go up in such a short time!

cc @mintsoft @jagtalon

**Comment by dhuette13** [comment]
Thanks! It would be neat to see it go live, let me know of any improvements I need to make.

File: lib/DDG/Goodie/LinuxFounder.pm [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+name "LinuxFounder";
+description "Answers the question of who is the founder of linux";
+primary_example_queries "linux founder", "founder of linux";
+secondary_example_queries "founder of linux operating system", "who is the founder of linux operating system";
+category "q/a";
+topics "geek";
+code_url "https://github.com/duckduckgo/zeroclickinfo-goodies/blob/master/lib/DDG/Goodie/LinuxFounder.pm";
+attribution github => ["dhuette13", "Daniel Huette"];
+
+# Triggers
```
Comment (javathunderman) [inline lib/DDG/Goodie/LinuxFounder.pm]: You can remove some of these guiders. (The trigger, metadata, and other comments). Don't remove the abstract, though.

File: t/LinuxFounder.t [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    
+    'who is the founder of linux operating system' => test_zci($answer, 
+        structured_answer => {
+            input => ["Linux Founder"],
+            operation => "Answer",
+            result => "Linus Torvalds"
+        }
+    ),
+    
+    # Bad Tests
```
Comment (javathunderman) [inline t/LinuxFounder.t]: I think you can get rid of these. ;)

**Comment by javathunderman** [comment]
@dhuette13 Left a few line notes. Other than that I think this can be merged. 

cc @MrChrisW @mintsoft @jagtalon

**Comment by moollaza** [comment]
@dhuette13 Congrats on your first Goodie! This is looking pretty good so far, but I just wanted to let you know that really the idea behind Instant Answers is that they should be a little more genera, rather than providing a single answer to a single question.

For example, one way to really improve this IA would be to generalize it so that it provides the name of the founder(s) for other operating systems as well. So queries like "Creator of Windows XP" or "Founder of Debian" could be answered. If you have a data source somewhere that can be parsed or manually converted into a Perl Hash for a lookup you could generalize your code a bit and then this IA would serve a much large query space.

Would you be interested in doing something like that?

**Comment by mintsoft** [comment]
@moollaza on a related topic I've just noticed if you search for "Linux Founder" it actually comes back with Mr Torvalds wikipedia article as the top result!
torvalds

**Comment by moollaza** [comment]
@mintsoft yup I had noticed that. I still think it would be neat to have a definitive Instant Answer for these kinds of queries, where you're looking for a specific person that we (or Wikipedia) can answer. Entity detection is another big focus for DDG :)

**Comment by dhuette13** [comment]
Thanks for the feedback! @moollaza I get what you're saying about generalizing the goodie. I'll see if I can find a data source that would fit. Since wikipedia is already a top result, maybe it would be nice if an about tab from wikipedia appeared instead of just a simple answer. Like when you search "Linus Torvalds" this shows up:
screenshot from 2015-01-20 09 36 18

At least then it would be more useful by providing more information.

**Comment by jagtalon** [comment]
@dhuette13 It turns out that we can send things like "linux founder" to point to the actual Wikipedia IA! You'd have to add a redirect to Wikipedia itself and it would just work since we continuously crawl them. More details in https://en.wikipedia.org/wiki/Wikipedia:Redirect.

As a result, the screenshot in https://github.com/duckduckgo/zeroclickinfo-goodies/pull/930#issuecomment-70674390 would pop up instead of a simple answer.

**Comment by moollaza** [comment]
ping! Hey @dhuette13 , I'm just checking in to see if you're still interested in finishing this one up? You could generalize the IA, or if you'd just like to address the query "linux founder" it would be best if you added a redirect to Wikipedia, in which case we won't need this IA anymore.

Please let us know what you're thinking, there's no rush either way :smiley:

**Comment by dhuette13** [comment]
Hey guys, sorry for the inactivity. I just started a school semester so time is tight. I like the idea of a redirect but I'm not sure how to put it into an instant answer. Also, I was unable to find a good data source for operating systems to generalize the IA. Feel free to close this request as I likely wont be able to work on it frequently. Though I would like to contribute more in the future! Thank you.

**Comment by mintsoft** [comment]
@dhuette13 no problem; I'll close this and mark as abandoned, if you want to re-pick it up at any point just reopen it :+1:

**Comment by moollaza** [comment]
>I like the idea of a redirect but I'm not sure how to put it into an instant answer

@dhuette13 a redirect wouldn't require any changes to DuckDuckGo. It's actually something you would update on Wikipedia.com -- you're able to signify words/phrases that redirect to a certain page. Once that's done our internal Wiki code will detect it and handle that query accordingly!

**Comment by marianosimone** [comment]
@moollaza I've just redirected Linux founder in Wikipedia to Linus Torvalds to see how this works (search !wiki linux founder to try it out!). Is that really all?

**Comment by moollaza** [comment]
@marianosimone awesome! Thanks a lot. We routinely update our Wiki data so soon enough searching "Linux Founder" should result in showing a Wikipedia IA result for "Linus Torvalds"

**Comment by javathunderman** [comment]
@moollaza I think we can remove the "Abandoned" label- I don't see any reason to keep going on this, seeing as it's already been taken care of as a redirect.

**Comment by mintsoft** [comment]
@javathunderman @moollaza I agree, it's not abandoned per-se, it's more redundant

**Comment by moollaza** [comment]
:+1:

**Pull request closed without merging**